### PR TITLE
python310Packages.python-trovo: init at 0.1.5

### PR DESCRIPTION
**Pull request by WolfangAukang**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/python-trovo/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, requests
+}:
+
+buildPythonPackage rec {
+  pname = "python-trovo";
+  version = "0.1.4";
```
Comment (fabaff) [inline pkgs/development/python-modules/python-trovo/default.nix:9]:
```suggestion
  version = "0.1.4";

  disabled = pythonOlder "3.8";
```

see https://codeberg.org/wolfangaukang/python-trovo/src/branch/main/pyproject.toml#L19

Comment (WolfangAukang) [reply pkgs/development/python-modules/python-trovo/default.nix:9]: Done

**Comment by WolfangAukang** [comment]
@ofborg eval

**Comment by WolfangAukang** [comment]
@fabaff do you know why I'm getting the following error from the checks?

```
 checking Nixpkgs on aarch64-darwin
error: builder for '/nix/store/gp0k5j3qmg5a30q5l41zf9s5j0adwwf1-nixpkgs-release-checks.drv' failed with exit code 1;
       last 1 log lines:
       > checking Nixpkgs on aarch64-darwin
       For full logs, run 'nix log /nix/store/gp0k5j3qmg5a30q5l41zf9s5j0adwwf1-nixpkgs-release-checks.drv'.
Error: Process completed with exit code 1.
```

**Review by fabaff — COMMENTED** [review]
The pinning of `requests` needs to be relaxed.

```bash
       > /build/python-trovo-0.1.4/dist /build/python-trovo-0.1.4
       > Processing ./python_trovo-0.1.4-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement requests==2.26.0 (from python-trovo) (from versions: none)
       > ERROR: No matching distribution found for requests==2.26.0
````

Would be best if that could be handled upstream.

**Comment by WolfangAukang** [comment]
Ready, I was able to fix the permissions requirements.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 156869` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.python-trovo</li>
    <li>python39Packages.python-trovo</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/156869#issuecomment-1059721488)
- [ ] all depending packages build
- [x] tested on x86_64

**Pull request merged**